### PR TITLE
Segfault fix for convolutional VAE example

### PR DESCRIPTION
**Pull request by hello-fri-end**

### CODE REVIEW

File: vae/vae_utils.hpp [hunk 1]
```diff
@@ -21,22 +21,23 @@ using namespace mlpack::ann;
...
          typename DataType = arma::mat>
-double MeanTestLoss(NetworkType model, DataType testSet, size_t batchSize)
+double MeanTestLoss(NetworkType& model, DataType& testSet, size_t batchSize)
 {
   double loss = 0;
   size_t nofPoints = testSet.n_cols;
   size_t i;
 
-  for (i = 0; i < (size_t)nofPoints / batchSize; i++)
+  for (i = 0; i < ( size_t )nofPoints / batchSize; i++)
```
Comment (zoq) [inline vae/vae_utils.hpp:30]:
```suggestion
  for (i = 0; i < (size_t) nofPoints / batchSize; ++i)
```

File: vae/vae_cnn_mnist/src/mnist_vae_cnn.cpp [hunk 2]
```diff
@@ -172,53 +185,54 @@ int main()
...
   for (int i = 0; i < cycles; i++)
   {
     // Train neural network. If this is the first iteration, weights are
     // random, using current values as starting point otherwise.
-    vaeModel.Train(train, train, optimizer);
+    vaeModel.Train(train,
+     		           train,
+     		           optimizer,
+     		           ens::PrintLoss(),
+     		           ens::ProgressBar());
```
Comment (zoq) [inline vae/vae_cnn_mnist/src/mnist_vae_cnn.cpp:226]:
```suggestion
    vaeModel.Train(train,
     		     train,
     		     optimizer,
     		     ens::PrintLoss(),
     		     ens::ProgressBar());
```

File: vae/vae_cnn_mnist/src/mnist_vae_cnn.cpp [hunk 3]
```diff
@@ -172,53 +185,54 @@ int main()
...
+                 // iteration.
+      0.9,       // Exponential decay rate for the first moment estimates.
+      0.999, // Exponential decay rate for the weighted infinity norm estimates.
+      1e-8,  // Value used to initialise the mean squared gradient parameter.
+      iterPerCycle, // Max number of iterations.
+      1e-8,         // Tolerance.
+      true);
+     
   const clock_t begin_time = clock();
+  clock_t cycle_time = begin_time;
```
Comment (zoq) [inline vae/vae_cnn_mnist/src/mnist_vae_cnn.cpp:214]: We should use camel-casing for all names.

**Review by zoq — APPROVED** [review]
Thanks, no more comments from my side.

File: vae/vae_generate.cpp [hunk 4]
```diff
@@ -39,14 +40,20 @@ int main()
...
 
   if (loadData)
   {
-    data::Load("mnist_full.csv", fullData, true, false);
+    data::Load("../data/mnist_train.csv", fullData, true, false);
+    // Get rid of the header
+    fullData =
+        fullData.submat(0, 1, fullData.n_rows - 1, fullData.n_cols -1);
     fullData /= 255.0;
+    // Get rid of the labels
```
Comment (kartikdutt18) [inline vae/vae_generate.cpp:54]:
```suggestion
    // Get rid of the labels.
```

**Review by kartikdutt18 — COMMENTED** [review]
Mentioned some minor style changes. Otherwise it looks good to me.

**Pull request merged**